osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -389,41 +388,42 @@ public override SnapResult SnapScreenSpacePositionToValidTime(Vector2 screenSpac
             return new SnapResult(screenSpacePosition, targetTime, playfield);
         }
 
-        public override float GetBeatSnapDistanceAt(double referenceTime)
+        public override float GetBeatSnapDistanceAt(HitObject referenceObject)
         {
-            DifficultyControlPoint difficultyPoint = EditorBeatmap.ControlPointInfo.DifficultyPointAt(referenceTime);
-            return (float)(100 * EditorBeatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier * difficultyPoint.SpeedMultiplier / BeatSnapProvider.BeatDivisor);
+            return (float)(100 * EditorBeatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier * referenceObject.DifficultyControlPoint.SliderVelocity / BeatSnapProvider.BeatDivisor);
```

Comment:
I have a slight feeling that doing this may break scrolling rulesets as those seem to be using the effect points' `ApproachRate` thing. Haven't tested, though.

Reply:
This is actually a member of `IPositionSnapProvider`, which is only used by osu! for positional / distance snap. So while it looks a bit weird being in such a central class as `HitObjectComposer`, it's correct implementation.